tests/test_operator_overloading.py

```diff
@@ -144,3 +145,10 @@ def test_overriding_eq_reset_hash():
 
         assert hash(hashable(15)) == 15
         assert hash(hashable(15)) == hash(hashable(15))
+
+
+def test_return_set_of_unhashable():
+    with pytest.raises(TypeError) as excinfo:
```

Comment:
Also optional: while we're at it, updating lines 139-140 accordingly would be nice. (I'd delete the then-redundant comment there.)

Reply:
@rwgk I did even better and added it as an explicit test.